storage/oasis/nodeapi/file/common.go

```diff
@@ -86,6 +86,8 @@ func GetFromCacheOrCall[Value any](cache KVStore, volatile bool, key CacheKey, v
 		if err2 != nil {
 			cache.logger.Warn(fmt.Sprintf("failed to unmarshal key %s from cache into %T: %v", key.Pretty(), result, err2))
 		}
+
+		return result, nil
```

Comment:
Shouldn't the non-cached value be fetched and returned in case errors happen during unmarshaling of the cached value (lines 82 and 87)?

Reply:
:heavy_plus_sign: : 
Plus I had another objection to my own code (a few lines up, if `cache.Has()` returns an error, we just errored out instead of going to the backing store). Then got carried away :disappointed:. Pushed a new commit. Andy, thank you for finding this and doing the initial fix. What a Wild West, I really should have added tests :/